libraries/dictionary/README.md

```diff
@@ -1,9 +1,261 @@
+| `range`    | `number`, `integer`   | [RangeRule](#rangerule-data-structure)                   | Maximum and minimum value conditions that a numeric field must pass.                                                                                                                                                                                                                                                                                                                                                           | `{ min: 5, exclusiveMax: 10 }` Represents an integer from 5-9.                                         |
+| `regex`    | `string`              | String (Regular Expression)                              | A regular expression pattern that the value must match.                                                                                                                                                                                                                                                                                                                                                                        | `^NCIT:C\d+$` Value must match an NCI Thesaurus ID.                                                    |
+| `value`    | all                   | Type of specified fields                                 | Field value exactly matches the value of the specified field. Strings are matched case sensitive.                                                                                                                                                                                                                                                                                                                              | `some_value`                                                                                           |
+
+### Meta Data Structure
+
+> **Meta Example**
+> ```json
+> {
+> 	"displayName": "Nicely Formatted Name",
+> 	"externalReferenceId": "ABCD:1234",
+> 	"exampleBooleanPropery": true,
+> 	"exampleNumericProperty": 123
+> }
+> ```
+
+A `meta` object is available to allow the dictionary creator to add custom properties to the Lectern Dictionary. The `meta` property is available to all Dictionary, Schema, and Field objects. Providing a `meta` value is optional, but if provided it must be a JSON object. There are no restrictions on the field names that can be added to the `meta` object other than they must be valid JSON. The values for meta properties are restricted to being a `string`, `number`, or `boolean` value. This means that meta data cannot be nested or accept arrays.
```

Comment:
`This means that meta data cannot be nested or accept arrays`.
Is this due to a technical limitation?

For PCGL currently this is the "workaround":
https://github.com/Pan-Canadian-Genome-Library/data-dictionary/blob/71fee8cc4cb2db40ec2e747f297a2ba9d2401e73/schemas/participant.json#L46C11-L46C148

Reply:
ok good feedback, string parsing is way worse... we were trying to keep this for simple usage and known property names... this shows you want to use it with any number of unknown properties. this can be changed I think.